### PR TITLE
[Doc] Refactor configuration doc and release config.get_argument_parser API

### PR DESCRIPTION
**Pull request by zhjwy9343**
*Issue #, if available:*

*Description of changes:*
This PR is the 1st one for the Notebook 6 PRs. This PR revises the model training and inference configuration document, moving the `--cf` argument out of launch argument set, and set it separately. In addition, this PR releases GraphStorm config module's `get_argument_parser` method for the Notebook 6, which will convert customized models that use GraphStorm APIs to use CLIs.

The preview pages include:
1. New configurations page: https://james4graphstorm.readthedocs.io/en/james_nb6_docs/cli/model-training-inference/configuration-run.html
2. New API index page: https://james4graphstorm.readthedocs.io/en/james_nb6_docs/api/references/index.html
3. Config API pages: https://james4graphstorm.readthedocs.io/en/james_nb6_docs/api/references/graphstorm.config.html and https://james4graphstorm.readthedocs.io/en/james_nb6_docs/api/generated/graphstorm.config.get_argument_parser.html#graphstorm.config.get_argument_parser  

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/source/api/references/graphstorm.config.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
+.. _apiconfig:
+
+graphstorm.config
+===================
+
+GraphStorm model training and inference CLIs have a set of :ref:`built-in configurations <configurations-run>`.
+These configurations are defined and managed in ``graphstorm.config`` module whose ``get_argument_parser()``
+method can help users to read in these built-in configurations either from a yaml file or from CLI arguments.
```
Comment (classicsong) [inline docs/source/api/references/graphstorm.config.rst:8]:
```suggestion
method can help users to load these built-in configurations either from a yaml file or from CLI arguments.
```
```suggestion
method can help users to read in these built-in configurations either from a yaml file or from CLI arguments.
```

File: python/graphstorm/config/argument.py [hunk 2]
```diff
@@ -76,7 +76,35 @@
 ]
 
 def get_argument_parser():
-    """Parse command-line arguments"""
+    """ Get GraphStorm CLIs argument parse.
```
Comment (classicsong) [inline python/graphstorm/config/argument.py:79]:
```suggestion
    """ Get GraphStorm CLI argument parse.
```
```suggestion
    """ Get GraphStorm CLIs argument parse.
```

File: python/graphstorm/config/argument.py [hunk 3]
```diff
@@ -76,7 +76,35 @@
...
+        
+        if __name__ == '__main__':
+            # use GraphStorm argument parser to accept configuration yaml file and other arguments
+            arg_parser = get_argument_parser()
+       
+            # parse all arguments and split GraphStorm's built-in arguments from the customized ones
+            gs_args, unknown_args = arg_parser.parse_known_args()
+       
+            print(f'GS arguments: {gs_args}')
+            print(f'Non GS arguments: {unknown_args}')
```
Comment (classicsong) [inline python/graphstorm/config/argument.py:101]: Shall we also demostrate how to create a GSConfig object?

File: docs/source/api/references/graphstorm.config.rst [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+This ``get_argument_parser()`` is useful when users want to convert your customized models to use GraphStorm CLIs.
```
Comment (jalencato) [inline docs/source/api/references/graphstorm.config.rst:9]:
```suggestion
This ``get_argument_parser()`` is useful when users want to convert customized models to use GraphStorm CLIs.
```

File: docs/source/api/references/graphstorm.config.rst [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+.. currentmodule:: graphstorm.config
```
Comment (jalencato) [inline docs/source/api/references/graphstorm.config.rst:11]: Shall we add a code example here to demonstrate how to use this function?

Comment (zhjwy9343) [reply docs/source/api/references/graphstorm.config.rst:11]: The code example is in the `get_argument_parser` own doc string.

File: python/graphstorm/config/argument.py (hunk 3 above)
Comment (zhjwy9343) [reply python/graphstorm/config/argument.py:101]: Yes. And then we need to release the GSConfig API too. Let me create another PR to release it.

File: docs/source/cli/model-training-inference/configuration-run.rst [hunk 6]
```diff
@@ -19,19 +17,26 @@ GraphStorm's model training and inference CLIs (both task-specific and task-agno
...
+.........................
+- **cf** or **yaml-config-file**: (**Required**) Path to the YAML configuration file.
+
 ..  note:: Below configurations can be set either in a YAML configuration file or be added as arguments of launch command.
 
 Environment Configurations
-----------------------------
-- **backend**: (**Required**) PyTorch distributed backend, the suggested backend is gloo. Support backends include gloo and nccl
+...........................
+- **backend**: (**Required**) PyTorch distributed backend, the suggested backend is `gloo`. Support backends include `gloo` and `nccl`.
```
Comment (jalencato) [inline docs/source/cli/model-training-inference/configuration-run.rst:39]: I just double check it here, the backend actually is not required, we return gloo as default. https://github.com/awslabs/graphstorm/blob/60edf9b907f5d0a5a29745c9bf43daa5c190798a/python/graphstorm/config/argument.py#L837

Comment (zhjwy9343) [reply docs/source/cli/model-training-inference/configuration-run.rst:39]: removed.

**Review by jalencato — APPROVED** [review]

**Pull request merged**